### PR TITLE
Glue mixtures

### PR DESCRIPTION
**Pull request by jinoobaek-qz**
Allow glue with weights

### CODE REVIEW

File: t5/data/mixtures.py [hunk 1]
```diff
@@ -26,24 +26,61 @@
...
     "super_glue_wsc_v102_simple_eval",
 ]
+
 _super_glue_tasks = _wsc_dpr_tasks + [
     "super_glue_%s_v102" % b.name
     for b in tfds.text.super_glue.SuperGlue.builder_configs.values()
     if "wsc" not in b.name
 ]
 
+_super_glue_tasks_with_weight = [(name, _SUPER_GLUE_WEIGHT_MAPPING[name]) for name in _super_glue_tasks]
```
Comment (adarob) [inline t5/data/mixtures.py:82]: let's use these on the proportional tasks and remove rate_num_examples

Comment (jinoobaek-qz) [reply t5/data/mixtures.py:82]: Added change. Is this what you meant?

File: t5/data/mixtures.py [hunk 2]
```diff
@@ -106,6 +142,12 @@
     ["squad_v010_allanswers"],
     default_rate=rate_num_examples)
 
+MixtureRegistry.add(
```
Comment (adarob) [inline t5/data/mixtures.py:145]: is this something new you added?

Comment (jinoobaek-qz) [reply t5/data/mixtures.py:145]: Yeah, I can remove this.

File: t5/data/mixtures.py [hunk 3]
```diff
@@ -26,24 +26,61 @@
...
+    'glue_mnli_v002': 392_702.,
+    'glue_qnli_v002': 104_743.,
+    'glue_rte_v002': 2_490.,
+    'glue_mnli_mismatched_v002': 0.,
+    'glue_mnli_matched_v002': 0.,
+    'glue_ax_v002': 0.,
+}
+
 # We omit WNLI because we train on WSC/DPR simple instead
 _glue_tasks = [
```
Comment (adarob) [inline t5/data/mixtures.py:59]: How about just doing?
_glue_tasks = _GLUE_WEIGHT_MAPPING.keys()?
_glue_tasks_with_weight = list(_GLUE_WEIGHT_MAPPING.items())

and ditto for superglue

File: t5/data/mixtures.py [hunk 4]
```diff
@@ -26,23 +26,47 @@
 import tensorflow_datasets as tfds
 
 
+_SUPER_GLUE_WEIGHT_MAPPING = {
```
Comment (craffel) [inline t5/data/mixtures.py:29]: Suggest adding comment stating that these mixture weights are based on the number of examples in each dataset.

File: t5/data/mixtures.py [hunk 5]
```diff
@@ -106,6 +129,7 @@
     ["squad_v010_allanswers"],
     default_rate=rate_num_examples)
 
+
```
Comment (craffel) [inline t5/data/mixtures.py:132]: Nit but I think you'll need to remove this space to avoid a linter error.

File: t5/data/mixtures.py [hunk 6]
```diff
@@ -25,24 +25,49 @@
 from t5.data.utils import rate_unsupervised
 import tensorflow_datasets as tfds
 
-
+# These weights are based on the number of examples in each dataset.
+_SUPER_GLUE_WEIGHT_MAPPING = {
+    "dpr_v001_simple": 1_322.,
+    "super_glue_wsc_v102_simple_train": 554.,
```
Comment (craffel) [inline t5/data/mixtures.py:31]: This number is based on the number of examples before preprocessing. Training examples from WSC get dropped when the target is not True (https://github.com/google-research/text-to-text-transfer-transformer/blob/master/t5/data/preprocessors.py#L1388). The correct number of examples here (after preprocessing) should be 259.

File: t5/data/mixtures.py [hunk 7]
```diff
@@ -25,24 +25,49 @@
...
+    "super_glue_wsc_v102_simple_eval": 104.,
```
Comment (craffel) [inline t5/data/mixtures.py:32]: WSC simple eval has no train split so it's probably clearer to set this to 0. https://github.com/google-research/text-to-text-transfer-transformer/blob/master/t5/data/tasks.py#L273

**Pull request merged**